hazelcast-integration/spring-jcache/src/main/java/com/hazelcast/spring/jcache/SpringJCacheJavaConfig.java

```diff
@@ -0,0 +1,24 @@
+package com.hazelcast.spring.jcache;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.core.Hazelcast;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+import static com.hazelcast.spring.jcache.DummyBeanTest.*;
+
+/**
+ * Example of Spring Caching with JCache
+ */
+public class SpringJCacheJavaConfig {
+    public static void main(String[] args) {
+        System.setProperty("hazelcast.jcache.provider.type", "client");
```

Comment:
`SpringHazelcastCachingProvider` handles this. It automatically detects if the instance is client or server and creates required type of caching provider.

I am not even sure if setting the property is neccessary for annotation based configuration. Let me check it.